miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -185,12 +194,16 @@ public BoxableView getBoxableView(BoxableId id) throws IOException {
   @Override
   public long save(Box box) throws IOException {
     setChangeDetails(box);
+    if (box.getStorageLocation() != null) {
+      box.setStorageLocation(storageLocationService.get(box.getStorageLocation().getId()));
+    }
     if (box.getId() == AbstractBox.UNSAVED_ID) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 32 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)